### PR TITLE
freetype2: work around GCC 12 ARM NEON bug

### PR DESCRIPTION
**Pull request by benoit-pierre**
Cf. https://gcc.gnu.org/bugzilla/show_bug.cgi?id=117811.

- - -
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader-base/1994)

### CODE REVIEW

**Comment by Frenzie** [comment]
Nice bug report :+1:

**Review by NiLuJe — APPROVED** [review]

File: thirdparty/freetype2/CMakeLists.txt [hunk 1]
```diff
@@ -1,4 +1,8 @@
-list(APPEND PATCH_FILES fix_meson_bzip2_option.patch)
+list(APPEND PATCH_FILES
+    fix_meson_bzip2_option.patch
+    # Work around GCC 12 ARM NEON bug (https://gcc.gnu.org/bugzilla/show_bug.cgi?id=117811).
+    ftcalc_gcc12_arm_neon_fix.patch
```
[the remark this answers is not in the document]
Comment (Frenzie) [reply thirdparty/freetype2/CMakeLists.txt:4]: Right, so the only negative effect is to spend a minute bit of time to do something that needn't be done with no eventual effect. I suppose we can live with that. :-)

**Review by Frenzie — APPROVED** [review]

**Comment by benoit-pierre** [comment]
I don't understand how the cache directory can change location mid build. ¯\\_(o.O)_/¯

**Comment by benoit-pierre** [comment]
Oh, wait, duh:
```bash
▹ grep build_scr .circleci/config.yml
173:          command: test -d build || source .ci/build_script.sh

▹ grep CCACHE .ci/build_script.sh
8:if [[ -z "${CCACHE_DIR}" ]]; then
9:    CCACHE_DIR="${HOME}/.ccache"
12:mkdir -p "${CCACHE_DIR}"
13:echo "using cache dir: ${CCACHE_DIR}"
22:    sudo chmod -R 777 "${CCACHE_DIR}"
24:        -v "${CCACHE_DIR}:${DOCKER_HOME}/.ccache" \
```

**Pull request merged**